vault/resource_quota_rate_limit.go

```diff
@@ -43,6 +43,19 @@ func quotaRateLimitResource() *schema.Resource {
 				Description:  "The maximum number of requests at any given second to be allowed by the quota rule. The rate must be positive.",
 				ValidateFunc: validation.FloatAtLeast(0.0),
 			},
+			"interval": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Description:  "The duration in seconds to enforce rate limiting for (default of 1).",
+				ValidateFunc: validation.IntAtLeast(0),
```

Comment:
An attempt to update `interval` to `0` results in Vault setting the value to `1`, arguably this should be an error.

E.g updating the first  test step's interval to 0:
```
=== RUN   TestQuotaRateLimit
    resource_quota_rate_limit_test.go:29: Step 1/5 error: Check failed: Check 4/5 error: vault_quota_rate_limit.foobar: Attribute 'interval' expected "0", got "1"
--- FAIL: TestQuotaRateLimit (0.95s)
```

```suggestion
				ValidateFunc: validation.IntAtLeast(1),
```